From 946255450d230ecf3d05ee575d7f1b69fd1c5713 Mon Sep 17 00:00:00 2001
From: Luwei Zhou <b45643@freescale.com>
Date: Mon, 11 Nov 2013 16:38:51 +0800
Subject: [PATCH 426/505] ENGR00287256 linux-test: Fix the SYNC mode hang
 issue when connected with MITB.

This patch fixes the hang and crash issue of MLB SYNC mode in the test.
The MITB will casue Sabreauto to hang and crash when testing the SYNC mode.
It is because MITB will cause something error on MLB bus when stopping the
SYNC test. The Sabreauto will keep entering error ISR and hang. Since we
don't know the details about MITB, we call IO_CTRL interface to disable
the interrupt in SYNC mode.

Signed-off-by: Luwei Zhou <b45643@freescale.com
---
 test/mxc_mlb_test/mxc_mlb150_test.c | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/test/mxc_mlb_test/mxc_mlb150_test.c b/test/mxc_mlb_test/mxc_mlb150_test.c
index a24b936..968f49c 100644
--- a/test/mxc_mlb_test/mxc_mlb150_test.c
+++ b/test/mxc_mlb_test/mxc_mlb150_test.c
@@ -291,6 +291,15 @@ int do_txrx_test(int fd)
 		return -1;
 	}
 
+	if (0 == t_case) {
+		ret = ioctl(fd, MLB_IRQ_DISABLE);
+		if (ret) {
+			printf("Failed to disable the IRQ\n");
+			return -1;
+		}
+	}
+
+
 	if (blocked) {
 		while (1) {
 			ret = read(fd, buf, 2048);
@@ -377,6 +386,15 @@ int do_txrx_test(int fd)
 
 shutdown:
 	usleep(1000);
+
+	if (0 == t_case) {
+		ret = ioctl(fd, MLB_IRQ_ENABLE);
+		if (ret) {
+			printf("Failed to enalbe the IRQ\n");
+			return -1;
+		}
+	}
+
 	ret = ioctl(fd, MLB_CHAN_SHUTDOWN);
 	if (ret) {
 		printf("Failed to shutdown async channel\n");
-- 
1.8.0

